### PR TITLE
REF: dont support dt64tz in nanmean

### PR DESCRIPTION
**Pull request by jbrockmendel**
nanmean is annotated as taking a ndarray, no reason for it to support dt64tz

### CODE REVIEW

File: pandas/core/nanops.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def _na_ok_dtype(dtype: DtypeObj) -> bool:
     return not issubclass(dtype.type, np.integer)
 
 
-def _wrap_results(result, dtype: DtypeObj, fill_value=None):
+def _wrap_results(result, dtype: np.dtype, fill_value=None):
     """ wrap our results if needed """
     if result is NaT:
```
Comment (jreback) [inline pandas/core/nanops.py:335]: can you add an assert here as well

**Review by jreback — CHANGES_REQUESTED** [review]
rebase as well

File: pandas/core/nanops.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/nanops.py:335]: what do you want to assert?

**Comment by jbrockmendel** [comment]
rebased+green

**Review by jreback — APPROVED** [review]

**Pull request merged**